app/helpers/metadata_helper.rb

```diff
@@ -62,6 +86,13 @@ def metadata
           { title: "Reset your password" }
         when "confirmations"
           { title: "Resend confirmation email" }
+        when "profiles"
+          case action_name
+          when "show"
+            { title: user_signed_in? && current_user == @user ? "My Profile" : "#{@profile.display_name}'s Profile" }
```

Comment:
You should be able to replace `user_signed_in? && current_user == @user` with `@user == current_user`